From 21ebe883ff82127da5ad9f2bac0123d63b9d706c Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Mon, 24 Mar 2025 21:56:24 +0000
Subject: [PATCH 2/9] Fail if BLIS is not found but `LINK_BLIS` is set

Fail if `LINK_BLIS` is set (defaults to `ON`) but the library is not
found. Insteaf of issung only a warning, return with a fatal error as
the configuration phase otherwise passes but building rocBLAS fails when
trying to link.
---
 clients/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 51d34060..33b71d0d 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -145,14 +145,14 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
       else()
         find_package(PkgConfig)
         if(NOT PKG_CONFIG_FOUND)
-          message( WARNING "Could not find libblis and pkgconfig is not available" )
+          message( FATAL_ERROR "Could not find libblis and pkgconfig is not available" )
         else()
           pkg_search_module(PKGBLAS cblas)
           if(PKGBLAS_FOUND)
             set( BLAS_LIBRARY ${PKGBLAS_LIBRARIES} )
             set( BLAS_INCLUDE_DIR ${PKGBLAS_INCLUDE_DIRS} )
           else()
-            message( WARNING "Could not find libblis and pkgconfig can not find any other implementation of cblas" )
+            message( FATAL_ERROR "Could not find libblis and pkgconfig can not find any other implementation of cblas" )
           endif()
         endif()
       endif()
-- 
2.45.1.windows.1

